modules/programs/khard.nix

```diff
@@ -0,0 +1,74 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.programs.khard;
+
+  accounts =
+    lib.filterAttrs (_: acc: acc.khard.enable) config.accounts.contact.accounts;
+
+  renderSettings = with lib.generators;
+    toINI {
+      mkKeyValue = mkKeyValueDefault rec {
+        mkValueString = v:
+          if lib.isList v then
+            lib.concatStringsSep ", " (map mkValueString v)
+          else if lib.isBool v then
+            if v then "yes" else "no"
+          else
+            v;
+      } "=";
+    };
+in {
+  meta.maintainers = [ lib.hm.maintainers.olmokramer ];
+
+  options.programs.khard = {
+    enable = lib.mkEnableOption "Khard: an address book for the Unix console";
+
+    settings = lib.mkOption {
+      type = with lib.types;
+        submodule {
+          freeformType = let
+            prim = either bool str;
+            primOrList = either prim (listOf prim);
```

Comment:
I don't think khard has any config options which "allow" bools in them, listOf str would suffice.

Reply:
Good point. I've changed the type to `oneOf [bool str (listOf str)]` now.